src/Microsoft.AspNetCore.Razor.Language/ITagHelperDescriptorProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public interface ITagHelperDescriptorProvider : IRazorEngineFeature
```

Comment:
This is the new API. You can have many providers and they are ordered. Because there's a context object with a 'results' collection later providers can modify the results of earlier ones.